src/endpoint/s3/s3_utils.js

```diff
@@ -785,6 +785,20 @@ function key_marker_to_cont_tok(key_marker, objects_arr, is_truncated) {
     return Buffer.from(j).toString('base64');
 }
 
+/**
+ * invalid_schema_to_aws_error handles specific AWS error responses related to invalid schema or parameters
+ * If the error's `rpc_code` is either `INVALID_SCHEMA` or `INVALID_SCHEMA_PARAMS`,
+ * the function throws a `MalformedPolicy` error. Otherwise, it throws the original error.
+ *
+ * @param {Object} error - The error object to handle
+ */
+function invalid_schema_to_aws_error(error) {
```

Comment:
Just to be on the safe side, you can also throw the error if (someone used this function improperly) `error.rpc_code` is `undefined`.

Reply:
Sure, Thanks